Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -542,6 +553,46 @@ QualifiedName MarkupAccumulator::xmlAttributeSerialization(const Attribute& attr
     return prefixedName;
 }
 
+LocalFrame* MarkupAccumulator::frameForAttributeReplacement(const Element& element)
+{
+    if (inXMLFragmentSerialization() || m_replacementURLStrings.isEmpty())
+        return nullptr;
+
+    auto* currentElement = const_cast<Element*>(&element);
+    if (!is<HTMLIFrameElement>(currentElement))
+        return nullptr;
+
+    auto& iframeElement = downcast<HTMLIFrameElement>(*currentElement);
```

Comment:
How about frameset?
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/frameset

Reply:
The link says not recommended, and I think we usually don't add support for to-be-deprecated interfaces. But maybe we do want to ensure the existence of frameset does not fail the saving; I will check on that.

Reply:
I don't think whether a feature is deprecated or not has much bearing in terms of supporting it in the serialization. The fact of matter is this feature is used in wild and it's going to be confusing for users if the feature didn't work with frameset.

Reply:
That's reasonable; I will follow up on <frame> support.